climada/engine/supplychain.py

```diff
@@ -0,0 +1,316 @@
+            production value of each country/sector-pair, i.e. each sector's
+            total production per country.
+        mriot_type (str): short string describing the mriot used for analysis.
+        cntry_pos (dict): dict with positions of all countries in the
+            mriot table and output arrays.
+        cntry_dir_imp (list): list with countries with direct impacts.
+    Attributes storing results of risk calculations:
+        years (np.array): 1-dim np.array containing all years for which impact
+            calculations where made (in yyyy format).
+        direct_impact (np.array): 2-dim np.array containing an impact-YEAR-set
+            with direct impact per year on each country/sector-pair.
+        direct_aai_agg (np.array): 1-dim np.array containing the average annual
+            direct impact for each country/sector-pair.
+        indirect_impact (np.array): 2-dim np.array containing an impact-YEAR-set
+            with indirect impact per year on each country/sector-pair.
+        indirect_aai_agg (np.array): 1-dim np.array containing the average annual
+            indirect impact for each country/sector-pair.
+        total_impact (np.array): 2-dim array containing an impact-year-set with
+            total (i.e. sum direct+indirect) impact per year on each
+            country/sector-pair.
+        total_aai_agg (np.array): 1-dim np.array containing the average annual
+            total impact for each country/sector-pair.
+        io_data (dict): dictionary with four key:value-pairs:
+            coefficients (np.array): 2-dim np.array containing the technical or
+                allocation coefficient matrix, depending on employed io approach.
+            inverse (np.array): 2-dim np.array containing Leontief or Ghosh
+                inverse matrix, depending on employed io approach.
+            io_approach (str): string informing about which io approach was
+                used in calculation of indirect risk.
+            risk_structure (np.array): 3-dim np.array containing for each year
+                the risk relations between all sector/country-pairs.
+        """
+    def __init__(self):
+        """Initialization"""
+        self.mriot_data = np.array([], dtype='f')
+        self.countries_iso3 = np.array([], dtype='str')
+        self.sectors = np.array([], dtype='str')
+        self.total_prod = np.array([], dtype='f')
+        self.mriot_type = 'None'
+        self.cntry_pos = {}
+
+    def read_wiod(self, year=2014, file_folder=SYSTEM_DIR.joinpath('results')):
+        """Read multi-regional input-output table of the WIOD project.
+        See www.wiod.org and the following paper: Timmer, M. P., Dietzenbacher,
+        E., Los, B., Stehrer, R. and de Vries, G. J. (2015), "An Illustrated
+        User Guide to the World Input–Output Database: the Case of Global
+        Automotive Production", Review of International Economics., 23: 575–605
+
```

Comment:
Since the method is to a large extent hard coding and dealing with the Release 2016 exclusively, I'd suggest to change the name into `read_wiod_2016` or something.

Reply:
mhh I find that a bit misleading as one could think is the WIOD table of year 2016, maybe something like wiod16

Reply:
updated